apps/kafka.go

```diff
@@ -0,0 +1,166 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
+type MetricsReceiverKafka struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+
+	confgenerator.MetricsReceiverSharedJVM `yaml:",inline"`
+
+	CollectJVMMetrics *bool `yaml:"collect_jvm_metrics"`
```

Comment:
See a related [comment](https://github.com/GoogleCloudPlatform/ops-agent/pull/349#discussion_r799868324) in https://github.com/GoogleCloudPlatform/ops-agent/pull/349

The logic around `CollectJVMMetrics` can be moved to a shared util and shared by Tomcat, Apache, Kafka and ElasticSearch (or any other applications that need this flag), so that the behavior is consistent, and in the future we don't need to worry about details of this flag.

Reply:
@qingling128 since that refactor was done in elasticsearch, I'm planning to either integrate it after elasticsearch is merged, or do it in a quick follow up PR to standardize on the refactored component. Does that make sense to you?

Reply:
Refactored to use the changes in elasticsearch as it was merged.

Reply:
Thanks!